edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.CloudProxy.Test/CloudProxyTest.cs

```diff
@@ -222,7 +222,7 @@ static async Task CheckMessageInEventHub(IList<IMessage> sentMessages, DateTime
             var eventHubReceiver = new EventHubReceiver(eventHubConnectionString);
             var cloudMessages = new List<EventData>();
             bool messagesFound = false;
-            // Add retry mechanism to make sure all the messages sent reached Event Hub. Retry 3 times.
+            // Add retry mechanism to make sure all the messages sent reached Event Hub.
```

Comment:
Is there a reason why this is a retry, and not a Task.Delay?

Reply:
eventHubReceiver.GetMessagesFromAllPartitions already did Task.Delay and it also in Line 235.  Therefore original dev want to retry after some time (20s) again.  This is my best guess why original dev did this.